File: packages/jsii-dotnet-runtime/src/Amazon.JSII.Runtime/JsiiException.cs
using System;
using Amazon.JSII.JsonModel.Api.Response;

namespace Amazon.JSII.Runtime
{
    public class JsiiException : Exception
    {
        public ErrorResponse ErrorResponse { get; }

        internal JsiiException(string message) : base(message)
        {
        }

        internal JsiiException(string message, Exception innerException)
            : base(message, innerException)
        {
        }

        internal JsiiException(ErrorResponse response, Exception innerException)
            : base(response.Error, innerException)
        {
            ErrorResponse = response;
        }
    }
}

File: packages/jsii-java-runtime/project/src/main/java/software/amazon/jsii/JsiiException.java
package software.amazon.jsii;

/**
 * An error raised by the jsii runtime.
 */
public final class JsiiException extends RuntimeException {
    public static final long serialVersionUID = 1L;

    /**
     * Creates an exception.
     * @param message The error message
     */
    JsiiException(final String message) {
        super(message);
    }

    /**
     * Creates an exception.
     * @param e The error that caused this exception
     */
    JsiiException(final Throwable e) {
        super(e);
    }

    /**
     * Creates an exception.
     * @param message The error message
     * @param e The error that caused this exception
     */
    JsiiException(final String message, final Throwable e) {
        super(message, e);
    }
}
